lalrpop/src/api/test.rs

```diff
@@ -0,0 +1,168 @@
+use std::env::{current_dir, set_current_dir, set_var, temp_dir};
+use std::fs;
+use std::path;
+use std::sync::{LockResult, Mutex, MutexGuard};
+
+use super::*;
+
+// tests may be run in parallel and these tests affect global state, so lock
+static API_TEST_MUTEX: Mutex<i32> = Mutex::new(0);
```

Comment:
I'm on board with using a Mutex here to make sure the tests are somewhat isolated. I was a little concerned with all of the panic'ing while holding a mutex(since I don't often reach for Mutex myself), but I imagine if there were issues with mutex poisoning, the test suite is failing anyways so it doesn't matter too much.

I will note that the comments I found also recommended https://docs.rs/serial_test/latest/serial_test/ as a dev dependency but also say that a mutex is completely reasonable.

Reply:
The `serial_test` crate looks cool to me. I didn't come across that when looking into it.  That would probably simplify the code a bit.  I vote that we not block this PR over that, but keep it in mind for a future enhancement.